Trivago/Jade/Application/Framework/JadeBundle/Controller/JsonApiController.php

```diff
@@ -493,14 +494,17 @@ protected function checkAccess($resourceName, $action)
             case ResourceConfig::ACTION_UPDATE:
                 $role = $resourceConfig->getUpdateRole();
                 break;
+            case ResourceConfig::ACTION_DELETE:
+                $role = $resourceConfig->getDeleteRole();
+                break;
             case ResourceConfig::ACTION_READ:
                 $role = $resourceConfig->getReadRole();
                 break;
             default:
                 throw new \LogicException('Invalid action passed');
         }
 
-        $this->denyAccessUnlessGranted($role, null, 'Unable to access this api!');
+        $this->denyAccessUnlessGranted($role, $entity, 'Unable to access this api!');
```

Comment:
@JuanWilde Not sure if this is actually going to be a major change or not
If I setup a simple project and say anyone user with ROLE_READ can get any objects and I don't put any voter what does denyAccessUnlessGranted returns?

Reply:
As I see If you don't put any voter the library will work as always since `$entity` is null by default. The problem is if we always pass `null` in the entity param, we are not able to check some stuff.. This is a working example of what I mean
```
    protected function voteOnAttribute($attribute, $subject, TokenInterface $token): bool
    {
        /** @var User|AnotherUser $user */
        $user = $token->getUser();

        if ($user instanceof Anotheruser) {
            return false;
        }

        if (self::READ === $attribute) {
            if ((null === $subject) || (null !== $subject && $subject->isOwnedBy($user))) {
                return true;
            }
        }

        if (self::CREATE === $attribute) {
            return true;
        }

        if (self::UPDATE === $attribute || self::DELETE === $attribute) {
            if ($subject->isOwnedBy($user)) {
                return true;
            }
        }

        return false;
    }
```

With you approach the subject is always null and you can not perform any action like check if the logged user is the owner... Makes sense?

Reply:
Ok. I tried the following thing. I have 2 entities, one goes throw a Voter and the other one not. The behaviour is the expected. I you try to get one object of the entity that doesn't have Voter, it works as soon as you define the ROLE_READ in the `jade.yml`. And for the other one, the same but passing throw the voter after.

I mean. With these changes, the library still working fine if you only use `jade.yml` and the listeners. But if you add a Voter it will work as well.